File: RedSheet.DbEntities/Models/vGeneratorContextModule.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vGeneratorContextModules", Schema = "dbo")]
	[Serializable]
	public partial class vGeneratorContextModule  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ApplicationModuleId { get; set; } 
 
	[Required]
	public bool IsRoot { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ModuleMasterId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string ModuleMasterName { get; set; } 

	public Nullable<int> ParentApplicationModuleId { get; set; } 

	public vGeneratorContextModule()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-generator-context-module.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vGeneratorContextModule {
    constructor(vGeneratorContextModule?: vGeneratorContextModule )  {
        let properties = [ "applicationModuleId", "isRoot", "moduleMasterId", "moduleMasterName", "parentApplicationModuleId",];
        for (let property of properties)
            if (vGeneratorContextModule && vGeneratorContextModule[property])
                this[property] = vGeneratorContextModule[property];
    }
 
	applicationModuleId : number =   0 ;
 
	isRoot : boolean = false ;
 
    @range(1,2147483647)
	moduleMasterId : number =   undefined;
 
    @required()
    @maxLength(100)
	moduleMasterName : string =   undefined;
 
	parentApplicationModuleId : number =   undefined;


}
